src/screens/ChannelListScreenV2/index.tsx

```diff
@@ -37,6 +42,81 @@ const ChannelListScreenV2 = () => {
     }
   };
 
+  const requestPermission = async () => {
+    const authStatus = await messaging().requestPermission();
+    const enabled =
+      authStatus === messaging.AuthorizationStatus.AUTHORIZED ||
+      authStatus === messaging.AuthorizationStatus.PROVISIONAL;
+
+    if (enabled) {
+      const fcmToken = await messaging().getToken();
+      const payload = {
+        fcm_token: fcmToken
+      };
+      fcmTokenService(payload);
+    }
+  };
+
+  React.useEffect(() => {
+    const checkNotif = async () => {
+      if (!navigation.isFocused()) return;
+      PushNotification.checkPermissions(
+        (
+          data: PushNotificationPermissions & {
+            authorizationStatus?: number;
+          }
+        ) => {
+          const lastPromptTime = StorageUtils.lastPromptNotification.get();
+
+          // For testing purpose,  (20000 milliseconds)
+          const promptInterval = 20000;
+          const currentTime = new Date().getTime();
+          if (lastPromptTime && currentTime - parseFloat(lastPromptTime) < promptInterval) {
+            return; // Don't show the prompt if the interval has not passed
+          }
+
+          const needsPermission =
+            data.alert === false || data.badge === false || data.sound === false;
+
+          const showAlert = (onPressAction: () => void) =>
+            Alert.alert(
+              "Don't Miss New Message",
+              'Allow notification to know when friends send you messages',
+              [
+                {text: 'Not now', onPress: () => console.log('Cancel Pressed'), style: 'cancel'},
+                {text: 'Allow', onPress: onPressAction}
+              ]
+            );
+
+          switch (data.authorizationStatus) {
+            case 1:
+              if (needsPermission) {
+                StorageUtils.lastPromptNotification.set(currentTime.toString());
+                showAlert(() => openSettings().catch(() => console.warn('cannot open settings')));
+              } else {
+                requestPermission();
+              }
+              break;
+            case 2:
+              if (needsPermission) {
+                StorageUtils.lastPromptNotification.set(currentTime.toString());
+                showAlert(() => openSettings().catch(() => console.warn('cannot open settings')));
+              }
+              break;
+            case 3:
+              StorageUtils.lastPromptNotification.set(currentTime.toString());
+              showAlert(() => requestPermission());
+              break;
+            default:
+              break;
+          }
+        }
+      );
+    };
+
+    checkNotif();
+  }, [navigation?.isFocused?.()]);
+
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `requestPermission` function and the `React.useEffect` hook are handling permissions for push notifications. However, there's no error handling in case of failures. For instance, if `messaging().getToken()` fails or `fcmTokenService(payload)` throws an error, it will result in an unhandled promise rejection.

Consider adding try-catch blocks to handle potential errors. Here's a suggestion:

```diff
  const requestPermission = async () => {
    try {
      const authStatus = await messaging().requestPermission();
      const enabled =
        authStatus === messaging.AuthorizationStatus.AUTHORIZED ||
        authStatus === messaging.AuthorizationStatus.PROVISIONAL;

      if (enabled) {
        const fcmToken = await messaging().getToken();
        const payload = {
          fcm_token: fcmToken
        };
        await fcmTokenService(payload);
      }
    } catch (error) {
      console.error('Error requesting permission:', error);
    }
  };

...

    checkNotif();
  }, [navigation?.isFocused?.()]);
```

Also, consider moving the `checkNotif` function outside of the `useEffect` hook to improve readability and maintainability. This way, you can also add error handling to this function.

```diff
+  const checkNotif = async () => {
+    if (!navigation.isFocused()) return;
+    try {
+      PushNotification.checkPermissions(
+        (
+          data: PushNotificationPermissions & {
+            authorizationStatus?: number;
+          }
+        ) => {
+          // ... existing code ...
+        }
+      );
+    } catch (error) {
+      console.error('Error checking permissions:', error);
+    }
+  };

-  React.useEffect(() => {
-    const checkNotif = async () => {
-      if (!navigation.isFocused()) return;
-      PushNotification.checkPermissions(
-        (
-          data: PushNotificationPermissions & {
-            authorizationStatus?: number;
-          }
+        ) => {
-          // ... existing code ...
-        }
-      );
-    };
-
    checkNotif();
  }, [navigation?.isFocused?.()]);
```

Lastly, the `console.warn` and `console.log` statements on lines 85 and 95 are not ideal for production code. Consider replacing them with proper error handling or logging mechanisms.


<!-- This is an auto-generated comment by OSS CodeRabbit -->